bin/propolis-server/src/lib/migrate/destination.rs

```diff
@@ -81,20 +82,41 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> DestinationProtocol<T> {
             .unwrap();
     }
 
-    async fn run(&mut self) -> Result<(), MigrateError> {
-        self.start();
-        self.sync().await?;
-        self.ram_push().await?;
-        self.device_state().await?;
-        self.arch_state().await?;
-        self.ram_pull().await?;
-        self.finish().await?;
-        self.end();
-        Ok(())
+    async fn run_phase(
+        &mut self,
+        step: MigratePhase,
+    ) -> Result<(), MigrateError> {
+        probes::migrate_phase_begin!(|| { format!("{:?}", step) });
```

Comment:
I might be inclined to `impl Display for MigratePhase` (instead of or in addition to deriving `Debug`) so that we directly control how the phase names are formatted instead of relying on the derive's implementation details (hard to foresee them changing, but *technically* they're [not stable](https://doc.rust-lang.org/std/fmt/trait.Debug.html#stability)...).

Reply:
Good thought -- I also felt like this was cheating a bit but wasn't super focused on the stability of the script. Implementing `Display` is easy to add though, so I can do that.